lib/stages/basebandReadout.cpp

```diff
@@ -171,149 +185,192 @@ void basebandReadout::main_thread() {
     }
 }
 
-void basebandReadout::readout_thread(const uint32_t freq_id, basebandReadoutManager& mgr) {
-    auto& request_no_data_counter = readout_counter.labels({std::to_string(freq_id), "no_data"});
+void basebandReadout::readout_thread(const uint32_t freq_ids[],
+                                     std::vector<basebandReadoutManager*>& mgrs) {
+    std::unique_ptr<basebandReadoutManager::requestStatusMutex> next_requests[_num_local_freq];
+    std::shared_ptr<basebandReadoutManager::requestStatusMutex> next_request;
+    kotekan::prometheus::Counter* request_no_data_counters[_num_local_freq];
```

Comment:
This was covered in more detail in slack, but these two arrays should likely just be converted to use vectors, since that's a safer access model anyway. And will get around the crazy MacOS clang bug.